### ISSUE TITLE
Codegen crash: fp32 broadcast along the stick (last) axis

### ISSUE THREAD

**Issue by moriohara** [torch-spyre/torch-spyre#4132]
# Codegen crash: fp32 broadcast along the stick (last) axis — `ddc_fold.cpp: innermostRelevantAllocLoop == innermostRelevantRefLoop`

## Summary

A multi-arg pointwise op that broadcasts an **fp32** operand along the **stick (last) axis** — i.e. the broadcaster is size-1 on the axis the device stick maps to, and the other operand is a full tensor on that axis — crashes in the ddc scheduler during codegen:

```
loc("…/bundle.mlir":…): error: sbf-ddc: DtException:
  innermostRelevantAllocLoop == innermostRelevantRefLoop,
  file …/deeptools/ddc/ddc_fold.cpp line 2747
error: sbf-run-scheduler-on-sdsc: failed on program 'sdsc_1'
```

surfaced to Python as a `subprocess.CalledProcessError` from `dxp_standalone`.

## This is EA-independent (not a mixed-EA issue)

The crash reproduces with a **pure-STANDARD** fp32 graph — no element-arrangement conversion involved:

```python
import torch, torch_spyre  # noqa
x = torch.randn(4, 1, dtype=torch.float32)   # broadcaster on the stick/last axis
w = torch.randn(4, 64, dtype=torch.float32)  # full on that axis
torch.compile(lambda a, b: a + b, dynamic=False)(x.to("spyre"), w.to("spyre"))  # -> ddc crash
```

Contrast (all pass):
- **fp16**, same shapes (`[4,1] + [4,64]`): ✅ works — fp16 stick-axis broadcast is supported.
- **fp32**, broadcast on a **non-stick** axis (`[1,64] + [4,64]`): ✅ works (max abs err ~2e-7).

So the gap is specifically **fp32 + broadcast along the stick axis**. Layouts (verified): the broadcaster is `dev_size=[1,4,32] stride_map=[-1,1,-1] coords=[0,d0,0]`; the full operand is `dev_size=[2,4,32] stride_map=[32,64,1] coords=[floor(d1/32),d0,Mod(d1,32)]`. The broadcaster must replicate its single stick value across the full operand's real 32-lane fp32 stick, which ddc fails to schedule.

## Relation to the mixed-EA broadcast change

PR #4110 (mixed-EA multi-arg pointwise) makes case 3.2 — a broadcastable *staggered* fp32 operand (e.g. an in-graph `.to(torch.float32)` upcast that is size-1 on the stick) combined with a STANDARD full operand — reach codegen, since a broadcastable staggered operand is physically identical to the STANDARD layout. That path then hits **this** crash. The PR's regression test `test_mixed_ea_staggered_broadcaster_fp32` asserts the failure is this codegen crash (not a layout-gate rejection) and should be converted to a passing `compare_with_cpu` once this issue is fixed.

## Repro / environment

- `tests/inductor/test_building_blocks.py::TestBuildingBlocks::test_mixed_ea_staggered_broadcaster_fp32`
- Or the pure-STANDARD snippet above.

## Suggested fix direction

Support fp32 stick-axis broadcast in the ddc scheduler / SDSC lowering (the fp16 path already works), so an fp32 size-1-stick operand can be replicated across a full fp32 stick.
